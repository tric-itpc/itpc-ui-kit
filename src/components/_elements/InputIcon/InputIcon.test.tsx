import React from "react"

import "@testing-library/jest-dom"
import { fireEvent, render, screen } from "@testing-library/react"

import { InputIcon } from "./index"

describe("Компонент InputIcon", () => {
  test("Рендерится с детьми (children)", () => {
    render(
      <InputIcon data-testid="input-icon">
        <span>Иконка</span>
      </InputIcon>
    )

    const iconElement = screen.getByTestId("input-icon")
    expect(iconElement).toBeInTheDocument()
    expect(iconElement).toHaveTextContent("Иконка")
  })

  test("Применяет класс itpc-input-icon", () => {
    render(
      <InputIcon data-testid="input-icon">
        <span>Иконка</span>
      </InputIcon>
    )

    const iconElement = screen.getByTestId("input-icon")
    expect(iconElement).toHaveClass("itpc-input-icon")
  })

  test("Корректно обрабатывает событие onClick", () => {
    const handleClick = jest.fn()
    render(
      <InputIcon data-testid="input-icon" onClick={handleClick}>
        <span>Иконка</span>
      </InputIcon>
    )

    const iconElement = screen.getByTestId("input-icon")
    fireEvent.click(iconElement)
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  test("Не вызывает onClick, если обработчик не передан", () => {
    const handleClick = jest.fn()
    render(
      <InputIcon data-testid="input-icon">
        <span>Иконка</span>
      </InputIcon>
    )

    const iconElement = screen.getByTestId("input-icon")
    fireEvent.click(iconElement)
    expect(handleClick).not.toHaveBeenCalled()
  })

  test("Рендерится без ошибок, если children не передан", () => {
    render(<InputIcon data-testid="input-icon" />)

    const iconElement = screen.getByTestId("input-icon")
    expect(iconElement).toBeInTheDocument()
    expect(iconElement).toBeEmptyDOMElement()
  })

  test("Рендерится корректно (snapshot)", () => {
    const { container } = render(
      <InputIcon data-testid="input-icon">
        <span>Иконка</span>
      </InputIcon>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
