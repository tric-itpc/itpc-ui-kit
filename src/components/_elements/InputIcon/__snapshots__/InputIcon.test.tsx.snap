// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент InputIcon Рендерится корректно (snapshot) 1`] = `
<span
  class="itpc-input-icon"
  data-testid="input-icon"
>
  <span>
    Иконка
  </span>
</span>
`;
